### ISSUE TITLE
The hardened frontmatter fallback rejects a double-quoted `name`, so the bundle fails its own validator wherever PyYAML is absent

### ISSUE THREAD

**Issue by agerescue** [rebelytics/one-skill-to-rule-them-all#197]
**Target:** `scripts/validate-skill-bundle.py`, `check_dir()` — the `ImportError`
branch that builds `data` without a parser, and `SKILL.md` line 2.
**Verified at:** upstream `main` (`80ab2f3`) and the v3.3.1 tag (`b20625d`) — the
relevant code and `SKILL.md` line 2 are identical in both. The install is a clean
copy of the tag with no local modifications (`diff -r` against the tag: no
differences outside `.github/`).
**Adjacent, not duplicate:** #158 (closed) is the parent — the same `ImportError`
branch, in the opposite direction. It reported the fallback *passing* a block a
real parser rejects; 3.3.0 hardened the fallback, and it now *rejects* a block
every real parser accepts. #146 (open) is about a header two parsers genuinely
disagree on; here they do not disagree — PyYAML and libyaml both accept the line,
and only the regex does not.

## What happened

Upgrading a 3.0.1 install to v3.3.1 and running the shipped validator on the
freshly installed bundle, on a machine where no `python3` on `PATH` has PyYAML:

```
FAIL:
  - frontmatter: `name` not kebab-case: '"task-observer"'
```

exit 1. The value is the skill's own, shipped at `SKILL.md` line 2:

```yaml
name: "task-observer"
```

## Reproduction

```bash
mkdir -p demo-skill demo-skill-bare
printf -- '---\nname: "demo-skill"\ndescription: "Use when a demonstration is needed of a double-quoted name value."\n---\n\n# Demo\n' > demo-skill/SKILL.md
printf -- '---\nname: demo-skill-bare\ndescription: "Use when a demonstration is needed of a bare name value."\n---\n\n# Demo\n' > demo-skill-bare/SKILL.md

python3 scripts/validate-skill-bundle.py demo-skill;      echo "exit=$?"   # FAIL, exit 1
python3 scripts/validate-skill-bundle.py demo-skill-bare; echo "exit=$?"   # exit 0
ruby -ryaml -e 'p YAML.safe_load(%q{name: "demo-skill"})'                  # {"name"=>"demo-skill"}
```

The two fixtures differ only in the quoting of `name`, so the fixture tests the
quotes and nothing else.

| Instrument | `name: "demo-skill"` | `name: demo-skill` |
|---|---|---|
| `validate-skill-bundle.py` at `b20625d`, `python3` without PyYAML | **FAIL, exit 1** | exit 0 |
| Ruby Psych 3.1.0 (libyaml), `YAML.safe_load` | `{"name"=>"demo-skill"}` | `{"name"=>"demo-skill"}` |
| The shipped v3.3.1 bundle, same interpreter | **FAIL, exit 1** | — |

The last row is the one that matters: the validator fails the bundle it ships
inside.

## Why the current text did not prevent it

```python
except ImportError:
    yaml_available = False
    data = {"name": (re.search(r"(?m)^name:\s*(.+)$", fm) or [None, ""])[1].strip(),
            "description": folded_description(fm)}
```

- `.strip()` removes whitespace, not quotes, so `NAME_RE` is applied to
  `"task-observer"` with the quote characters still in the string. A real parser
  never sees those characters as part of the value.
- The template asks for this line. 3.2.0's "frontmatter that parses" work made the
  template quote every prose value, and `SKILL.md` follows it. The fallback and the
  template now disagree about the skill's own header.
- The next branch would fail the same way for the same reason: `name != skill_dir.name`
  compares `'"task-observer"'` to `task-observer`. Only the kebab-case check fires
  first.
- `description` is unaffected only because it is length-checked rather than
  pattern-matched, so the same missing unquote does not surface there.
- On this platform the fallback is the default branch: neither `/usr/bin/python3`
  nor the Homebrew `python3` has PyYAML. That is the same platform condition #158
  reported, so the branch that regressed is the one that actually runs.

## Suggested fix

1. **Unquote in the fallback before any check.** In the `ImportError` branch, strip
   one layer of matching surrounding quotes from the captured value:

   ```python
   raw = (re.search(r"(?m)^name:\s*(.+)$", fm) or [None, ""])[1].strip()
   if len(raw) >= 2 and raw[0] == raw[-1] and raw[0] in "\"'":
       raw = raw[1:-1]
   ```

   A single-quoted value has the same problem and the same fix.

2. **Make the bundle its own planted control.** The repository ships an artefact
   that exercises this: itself. Running the validator against the bundle in
   *both* modes — with and without PyYAML importable — is a two-line check, and
   it is the control that would have caught this before the tag. A FAIL there is
   never a property of the input.

3. Optional, and separate from the defect: when the fallback reports a failure
   whose cause is a shape a parser would have accepted, the message reads as a
   verdict about the author's file. Naming the mode in the failure line
   (`frontmatter (pattern-checked, no YAML loader): ...`) keeps #158's "say which
   way it went" property on the failure path as well as the pass path.

## Principle

Hardening a fallback so it stops accepting what the consumer rejects can, in the
same change, make it reject what the consumer accepts; the fix closes one
direction and nothing measures the other. When the checker ships inside the
artefact it checks, that artefact is the cheapest available control for the
direction nobody tested — and a validator that fails its own bundle has measured
itself and reported the result as a property of the input.

**Comment by agerescue** [comment]
Verified after the fact, on the same machine and the same interpreter: with
PyYAML 6.0.3 installed, the shipped v3.3.1 bundle passes — `OK: all gate checks
passed`, exit 0 — and both fixtures pass. Nothing about the header changed, so
the defect is entirely in the `ImportError` branch.

| Input | `python3` without PyYAML | same `python3`, PyYAML 6.0.3 |
|---|---|---|
| the shipped v3.3.1 bundle | FAIL, exit 1 | **OK, exit 0** |
| `name: "demo-skill"` | FAIL, exit 1 | exit 0 |
| `name: demo-skill` | exit 0 | exit 0 |
| description wrapped onto an unindented line (#158's fixture) | — | **FAIL, exit 1** |

The last row is the control for the obvious objection — that a parser being
present just means a laxer gate. It does not: the block #158 reported still
fails with the parser present, and fails twice over, naming both the parse error
and the offending line.

```
FAIL:
  - frontmatter did not parse (likely an unquoted `: ` in a value — double-quote the value or reword it): while scanning a simple key
  - frontmatter:3: line is neither `key: value` nor an indented continuation
```

So on this input the fallback's quote handling is the only thing separating the
two modes, and 3.3.0's structural floor is doing its job on the input it was
written for.

One note on the workaround, since it is the obvious first move: installing
PyYAML does not fix this, it only stops the affected branch from running. The
report stands for every interpreter without it — which on this platform was all
of them.

**Comment by rebelytics** [comment]
Thank you very much, I'll take care of this as soon as possible!
